### PR TITLE
Possibility to use Chart.AppVersion field as image tag when it is not specified

### PR DESCRIPTION
**Pull request by oOraph**

### CODE REVIEW

File: charts/common/templates/_images.tpl [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-{{ include "hf.common.images.image" ( dict "imageRoot" .Values.path.to.the.image "global" .Values.global ) }}
+{{ include "hf.common.images.image" ( dict "imageRoot" .Values.path.to.the.image "global" .Values.global "appVersion" .Chart.AppVersion ) }}
 */}}
 {{- define "hf.common.images.image" -}}
 {{- $useGlobalRegistry := or (not (hasKey .imageRoot "useGlobalRegistry")) .imageRoot.useGlobalRegistry -}}
 {{- $registryName := .imageRoot.registry -}}
 {{- $repositoryName := .imageRoot.repository -}}
 {{- $separator := ":" -}}
-{{- $termination := .imageRoot.tag | toString -}}
+{{- $termination := .imageRoot.tag | default .appVersion | toString -}}
```
Comment (rtrompier) [inline charts/common/templates/_images.tpl]: Maybe just rename `.appVersion` by `.tag` to keep the same name.

File: charts/unit-tests/tests/deployment_test.yaml [hunk 2]
```diff
@@ -64,6 +64,17 @@ tests:
       - equal:
           path: spec.template.spec.containers[2].image
           value: spaces/app-manager:latest
+  - it: deployment should have the correct image tag reflectiing chart appVersion
```
Comment (rtrompier) [inline charts/unit-tests/tests/deployment_test.yaml]: Missing a line break before :)

File: charts/unit-tests/templates/_helpers.yaml [hunk 3]
```diff
@@ -7,9 +7,13 @@
 {{- end -}}
 
 {{- define "appManager.image" -}}
-{{ include "hf.common.images.image" (dict "imageRoot" .Values.images.appManager "global" .Values.global.huggingface) | quote }}
+{{ include "hf.common.images.image" (dict "imageRoot" .Values.images.appManager "global" .Values.global.huggingface "appVersion" .Chart.AppVersion) | quote }}
```
Comment (rtrompier) [inline charts/unit-tests/templates/_helpers.yaml]: No need to modify this exemple no ?

**Review by rtrompier — APPROVED** [review]
LGTM. Thanks :)

File: charts/common/templates/_images.tpl (hunk 1 above)
Comment (oOraph) [reply charts/common/templates/_images.tpl]: done

File: charts/unit-tests/tests/deployment_test.yaml (hunk 2 above)
Comment (oOraph) [reply charts/unit-tests/tests/deployment_test.yaml]: done

File: charts/unit-tests/templates/_helpers.yaml (hunk 3 above)
Comment (oOraph) [reply charts/unit-tests/templates/_helpers.yaml]: It proves that the .tag field is a fallback value, not used when the tag field is already specified in the image
This way we test the following cases:
Tag field and no fallback specified
No tag field in image but fallback specified 
Both specified
The only one that will break is: none specified (not tested)

Comment (oOraph) [reply charts/unit-tests/templates/_helpers.yaml]: So I reused one of the existing tests to check this. But if you prefer, we can indeed create an additional unittest

**Review by rtrompier — APPROVED** [review]

**Pull request merged**